homeassistant/components/upnp/__init__.py

```diff
@@ -119,13 +121,6 @@ async def async_setup_entry(hass: HomeAssistantType, config_entry: ConfigEntry)
     # Save device
     hass.data[DOMAIN][DOMAIN_DEVICES][device.udn] = device
 
-    # Ensure entry has proper unique_id.
```

Comment:
What if there's a config entry that was previously imported and lacks the correct unique_id? It will now be loaded and keep lacking a unique id, right?

Reply:
Yes, there is a small window between `async_step_import()` and this, `async_setup_entry()`, in the pre-PR version. I think the window is very small and doubt there are users who have this.

In the new version the `unique_id` is always set via the config flows.
`async_step_user()`: https://github.com/home-assistant/core/blob/f87c9247750af65512a58546b406f72e4c522d4d/homeassistant/components/upnp/config_flow.py#L60
`async_step_import()`: https://github.com/home-assistant/core/blob/f87c9247750af65512a58546b406f72e4c522d4d/homeassistant/components/upnp/config_flow.py#L138
`async_step_ssdp()`: https://github.com/home-assistant/core/blob/f87c9247750af65512a58546b406f72e4c522d4d/homeassistant/components/upnp/config_flow.py#L163

As far as I know there is no path to `async_setup_entry()` without either going through either an existing entry, where it is already set, or the config flows, where it is always set now.

If I'm wrong please let me know. Then I'll change it to something like:
```
    if not config_entry.unique_id:
        hass.config_entries.async_update_entry(
            entry=config_entry,
            unique_id=device.unique_id,
        )
```

Reply:
Before this PR we didn't set unique_id in the import step. So a config entry could have been created via import and then the config yaml removed. Then the config entry would upon loading not have the correct unique_id, from what I can see.

Reply:
Possible, but the config_entry is directly activated as far as I know. So the window would be fairly small.

Nevertheless, I've added my suggested snippet.

Reply:
I think this covers your case. Thank you for the review.

Reply:
Ok, I think you're correct that the old imported entry would receive the unique id here directly after creation.